### PR TITLE
Clarify the core dependencies mappings: bbolt and raft

### PR DESCRIPTION
**Pull request by ahrtr**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

cc @fuweid @jmhbnz @serathius @spzala

### CODE REVIEW

File: Documentation/contributor-guide/dependency_management.md [hunk 1]
```diff
@@ -126,3 +127,20 @@ For an example of how to update etcd to a new patch release of Go refer to issue
...
 
 - <https://github.com/kubernetes/sig-release/blob/master/release-engineering/handbooks/go.md>
+
+## Core dependencies mappings
+[bbolt](https://github.com/etcd-io/bbolt) and [raft](https://github.com/etcd-io/raft) are two core dependencies of etcd.
+
+Both etcd 3.4.x and 3.5.x depend on bbolt 1.3.x, and etcd 3.6.x (`main` branch) depends on bbolt 1.4.x.
+
+raft is included in etcd repository for release-3.4 and release-3.5 branches, so etcd 3.4.x and 3.5.x do not depend on any
+external raft module. We moved raft into [a separate repo](https://github.com/etcd-io/raft) starting from 3.6 (`main` branch), and the first raft
```
Comment (jmhbnz) [inline Documentation/contributor-guide/dependency_management.md:137]:
```suggestion
external raft module. We moved raft into [a separate repository](https://github.com/etcd-io/raft) starting from 3.6 (`main` branch), and the first raft
```

File: Documentation/contributor-guide/dependency_management.md [hunk 2]
```diff
@@ -126,3 +127,20 @@ For an example of how to update etcd to a new patch release of Go refer to issue
 References:
 
 - <https://github.com/kubernetes/sig-release/blob/master/release-engineering/handbooks/go.md>
+
+## Core dependencies mappings
```
[the remark this answers is not in the document]
Comment (ahrtr) [reply Documentation/contributor-guide/dependency_management.md:131]: fixed & thx

File: Documentation/contributor-guide/dependency_management.md [hunk 3]
```diff
@@ -126,3 +127,20 @@ For an example of how to update etcd to a new patch release of Go refer to issue
...
+[bbolt](https://github.com/etcd-io/bbolt) and [raft](https://github.com/etcd-io/raft) are two core dependencies of etcd.
+
+Both etcd 3.4.x and 3.5.x depend on bbolt 1.3.x, and etcd 3.6.x (`main` branch) depends on bbolt 1.4.x.
+
+raft is included in etcd repository for release-3.4 and release-3.5 branches, so etcd 3.4.x and 3.5.x do not depend on any
```
[the remark this answers is not in the document]
Comment (ahrtr) [reply Documentation/contributor-guide/dependency_management.md]: fixed

File: Documentation/contributor-guide/dependency_management.md (hunk 1 above)
Comment (ahrtr) [reply Documentation/contributor-guide/dependency_management.md]: updated

File: Documentation/contributor-guide/dependency_management.md [hunk 4]
```diff
@@ -126,3 +127,20 @@ For an example of how to update etcd to a new patch release of Go refer to issue
...
+external raft module. We moved raft into [a separate repo](https://github.com/etcd-io/raft) starting from 3.6 (`main` branch), and the first raft
+release will be v3.6.0, so etcd 3.6.x will depend on raft 3.6.x.
+
+Please see the table below,
```
[the remark this answers is not in the document]
Comment (ahrtr) [reply Documentation/contributor-guide/dependency_management.md]: updated.

**Comment by ahrtr** [comment]
> Few minor nitpicks on lint and grammar if you have time.

Thanks. All fixed.

**Review by fuweid — APPROVED** [review]
LGTM

**Pull request merged**